src/Pos/Update/Poll/Logic/Apply.hs

```diff
@@ -283,16 +283,22 @@ verifyBlockVersion upId UpdateProposal {..} = do
 -- Here we check that proposal has at least 'genesisUpdateProposalThd'
 -- stake of total stake in all positive votes for it.
 verifyProposalStake
-    :: (MonadError PollVerFailure m)
+    :: (MonadPollRead m, MonadError PollVerFailure m)
     => Coin -> [(UpdateVote, Coin)] -> UpId -> m ()
 verifyProposalStake totalStake votesAndStakes upId = do
-    let threshold = applyCoinPortion genesisUpdateProposalThd totalStake
+    let threshold =
+          coinToInteger $ applyCoinPortion genesisUpdateProposalThd totalStake
     let votesSum =
             sumCoins . map snd . filter (uvDecision . fst) $ votesAndStakes
-    when (coinToInteger totalStake < votesSum) $
+    logDebug $
+        sformat
+            ("Verifying stake for proposal "%shortHashF%
+             ", threshold is "%int%", voted stake is "%int)
+            upId threshold votesSum
+    when (votesSum < threshold) $
```

Comment:
Because coin is not `Integral` and I was too lazy to think, so I did what came into my mind first 😛 
Well, I think I'd rather fix it, ok.